modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -2168,12 +2168,29 @@ def map(
         PandasDataframe
             A new dataframe.
         """
-        map_fn = (
-            self._partition_mgr_cls.lazy_map_partitions
-            if lazy
-            else self._partition_mgr_cls.map_partitions
-        )
-        new_partitions = map_fn(self._partitions, func, func_args, func_kwargs)
+        if self.num_parts <= CpuCount.get():
```

Comment:
Maybe we shouldn't be so strict on the number of partitions. We have some heuristics in qc.dropna and mgr.rebalance_partitions. Could you try applying those here? Perf measurements would also be valuable.

Reply:
btw, should we use here `NPartitions` instead of `CpuCount`? @YarShev what do you think?

Reply:
I'm not sure because NPartitions can be changed and then the number of partitions can be much less than CpuCount. In this case, we don't need to use the full_axis approach because some workers will be free.

Reply:
Maybe we should check for partitions.shape[0] and [1] explicitly whether to use full_axis or not?